render/render.js

```diff
@@ -349,15 +349,7 @@ module.exports = function($window) {
 		}
 	}
 	function isRecyclable(old, vnodes) {
-		if (old.pool != null && Math.abs(old.pool.length - vnodes.length) <= Math.abs(old.length - vnodes.length)) {
-			var oldChildrenLength = old[0] && old[0].children && old[0].children.length || 0
-			var poolChildrenLength = old.pool[0] && old.pool[0].children && old.pool[0].children.length || 0
-			var vnodesChildrenLength = vnodes[0] && vnodes[0].children && vnodes[0].children.length || 0
-			if (Math.abs(poolChildrenLength - vnodesChildrenLength) <= Math.abs(oldChildrenLength - vnodesChildrenLength)) {
-				return true
-			}
-		}
-		return false
+		return old.pool != null && vnodes.length > 0 && vnodes[0] != null && (vnodes[0].key != null ? true : vnodes.length > old.length)
```

Comment:
That last ternary could be the equivalent (and simpler) `vnodes[0].key != null || vnodes.length > old.length`.

Also, could you explain to me why this is equivalent? I'm not 100% sure how it is, since I'm not quite as intimately familiar with the invariants leading up to this. Not saying it is or isn't, I'm just curious.